File: apps/be-java-maven/src/main/java/com/soo/apps/webfilesystem/dto/FileInfoDto.java
package com.soo.apps.webfilesystem.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FileInfoDto {

    private String name;
    private String path;
    private String type; // file, directory, symlink, junction
    private long size;
    private String extension;
    private LocalDateTime createdAt;
    private LocalDateTime modifiedAt;
    private boolean isReadable;
    private boolean isWritable;
    private boolean isExecutable;
    private boolean isHidden;
    private boolean isSymbolicLink;
    private boolean isJunctionPoint;
    private String linkTarget;
    private String parentPath;
    private boolean hasChildren;
}

File: apps/be-nestjs-ts/src/apps/webfilesystem/dto/file.info.dto.ts
// FileInfoDto.java에 해당하는 TypeScript DTO

export interface FileInfoDto {
    name: string;
    path: string;
    type: string; // file, directory, symlink, junction
    size: number;
    extension?: string;
    createdAt: string; // ISO string
    modifiedAt: string; // ISO string
    isReadable: boolean;
    isWritable: boolean;
    isExecutable: boolean;
    isHidden: boolean;
    isSymbolicLink: boolean;
    isJunctionPoint: boolean;
    linkTarget?: string;
    parentPath?: string;
}
